CDS_Dashboard/script.js

```diff
@@ -353,320 +346,145 @@ chart = new Highcharts.Chart({
-            }
-            return s;
         }
-    },
-    series: [{  // sets chart series here 
-        type: chartType,
-        name: name,
-        data: data,
-        color: 'white' // don't know why you set this to white
-    }],
+    }, //end of plotOptions
     exporting: {
         enabled: false // can't print chart
     }
-}); // end of chart = new Highcharts.Chart()
-
+});
 
-
-chart2 = new Highcharts.Chart({
-    credits: {
-        enabled: false
-    },
-    chart: {
-        style: {
-            fontFamily: 'SourceSansProRegular, Open Sans, Helvetica Neue, Arial, sans-serif'
+    var chart1 = {
+        chart: {
+            renderTo: 'chartContainer'
         },
-        renderTo: 'secondContainer', // passing it where to put it on the HTML - there is a div with ID 'chartContainer'
+        plotOptions: {
+            series: {
+                cursor: 'pointer', // cursor set to pointer since we have 'click' events
+                point: {
+                    events: {
+                        click: function () {
+                            var drilldown = this.drilldown; // uses keyword 'this'
+                            var options; 
+                            mq.addListener(responsiveHighCharts);
+                            responsiveHighCharts(mq);
+                            
+                            
+                            function responsiveHighCharts(mq) {
+                                if (mq.matches) {
+                                       // create variable to passed into setChart function
+                                        setTimeout(function() {
+                                            setSecondChart(options);
+                                        },700);
+                                        options = buildOptions(drilldown,options);
 
-    }, // end of chart property
-    title: {
-        text: title,
-        style: {
-            fontWeight: 'bold',
-            textTransform: 'uppercase'
-        }
-    },
-    subtitle: {
-        text: 'Click on Pie Graph Above to Learn More'
-    },
-    xAxis: {
-        categories: categories
-    },
-    yAxis: { // i don't know get this
-        title: {
-            text: 'Assignments'
-        }
-    },
-    plotOptions: {
-        pie: {
-            dataLabels: {
-                enabled: true,
-                //distance: -10,
-                color: 'black',
-                style: {
-                    fontWeight: 'bold'
+                                } else {
+                                    if (counter === 0) {
+                                        counter++;
+                                        moveChart();
+                                        setTimeout(function() {
+                                            setSecondChart(options);
+                                        },700);
+                                        options = buildOptions(drilldown, options);
```

Comment:
I have moved the var options declaration inside of the responsiveHighCharts function to make my code DRY and address your comment. Options is only built one time per click on the first graph - and what object it is pointing to (dependent upon the return of the buildOptions function).